Web Inspector: Add support for getting user agents from WebKitAdditions if present

patrickangle opened the pull request:
#### d25915a5a0c5cc5c2a0b1e2b38f4010a309b0419
<pre>
Web Inspector: Add support for getting user agents from WebKitAdditions if present
<a href="https://bugs.webkit.org/show_bug.cgi?id=252030">https://bugs.webkit.org/show_bug.cgi?id=252030</a>
rdar://96018617

Reviewed by Devin Rousso and Tim Horton.

Ports may desire to provide their own set of User Agents for Web Inspector to display for remote inspection via
WebKitAdditions.

* Source/WebInspectorUI/UserInterface/Base/Main.js:
(WI.loaded):
- Place an empty WebKitAdditions in the global scope if one doesn&apos;t already exist, since optional chaining can&apos;t be used
to check for a non-declared root object.
- Also some drive-by indentation fixes for the UAs.

Canonical link: <a href="https://commits.webkit.org/260335@main">https://commits.webkit.org/260335@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4d7a456788153bd163eb627ccf68038770657e96

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108041 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/17101 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40915 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/117168 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/116500 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/18472 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8410 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/100239 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113807 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/76/builds/18472 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/97152 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/41854 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/76/builds/18472 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28790 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83523 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/10001 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/30138 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10720 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/7039 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/16151 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49728 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7151 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12299 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebInspectorUI/UserInterface/Base/Main.js:
```diff
@@ -50,6 +50,9 @@ WI.LayoutDirection = {
 WI.loaded = function()
 {
+    // Ports may or may not include additions.
+    window.WebKitAdditions ??= {};
```
hmm kinda odd that this is inside `WI.loaded`

`WebKitAdditions/WebInspectorUI/WebInspectorUIAdditions.js` is already included at the end of `Main.html`, so I think it'd probably be easier to just have that file completely override whatever was previously declared instead of having this do a `??=`

as such, why dont we move this to be a `var WebKitAdditions = {};` inside `WebInspector.js` (i.e. alongside where `WI` is first declared)?

patrickangle committed:
Web Inspector: Add support for getting user agents from WebKitAdditions if present
https://bugs.webkit.org/show_bug.cgi?id=252030
rdar://96018617

Reviewed by Devin Rousso and Tim Horton.

Ports may desire to provide their own set of User Agents for Web Inspector to display for remote inspection via
WebKitAdditions.

* Source/WebInspectorUI/UserInterface/Base/Main.js:
(WI.loaded):
- Place an empty WebKitAdditions in the global scope if one doesn't already exist, since optional chaining can't be used
to check for a non-declared root object.
- Also some drive-by indentation fixes for the UAs.

Canonical link: https://commits.webkit.org/260335@main
